### DISCUSSION TITLE
Any way to send to voicemail after x minutes in Queue?

### DISCUSSION THREAD

**Discussion by jlaurie87** [amazon-connect/amazon-connect-salesforce-scv#107] [Q&A] [answered]
Hey there, We have voicemail express setup and working. 

I have an odd use case though. If an agent is available online to take an incoming call, the call is sent to the queue. If the agent doesnt accept the call and goes offline, the customer is left in waiting until another agent comes online. 

Do you happen to know if I can use VE2 to send the contact to the voicemail flow if they have waited X minutes? 

Cheers

**Comment by dougjaso** [comment]
Sure. You’d just need to look at the time in the queue flow and present that option as appropriate.

-----------
Jason Douglas
Principal Solutions Architect
Amazon Connect

628.336.1973 – mobile
415.262.9741 – office

[signature_1243584860]<http://www.aws.amazon.com/>


From: jlaurie87 ***@***.***>
Reply-To: amazon-connect/amazon-connect-salesforce-scv ***@***.***>
Date: Monday, June 6, 2022 at 1:31 PM
To: amazon-connect/amazon-connect-salesforce-scv ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [amazon-connect/amazon-connect-salesforce-scv] Any way to send to voicemail after x minutes in Queue? (Discussion #107)


Hey there, We have voicemail express setup and working.

I have an odd use case though. If an agent is available online to take an incoming call, the call is sent to the queue. If the agent doesnt accept the call and goes offline, the customer is left in waiting until another agent comes online.

Do you happen to know if I can use VE2 to send the contact to the voicemail flow if they have waited X minutes?

Cheers

**Reply by jlaurie87** [reply]
Thanks @dougjaso - Makes sense. I currently have the contact flow using omnichannel flow to route the calls, and I know the contact is hearing the default customer queue audio from that customer queue flow (holding music)

I just am unsure where to check the contacts current queue time and if it is over x minutes, send to voicemail. I know this is an extension of the support for this app, so its ok if you cant answer that. I guess I am just confused for where a contact is in a flow is the flow ends on a transfer to queue block.

**Reply by chrisansotti** [reply, marked as answer]
I was able to accomplish this in the "Default customer queue" contact flow. It should start with a Loop Prompts block where I just have hold music playing, but then there's an option to "Interrupt" after X number of minutes. When you check that you can then move the call to another block that transfers them to the Voicemail contact flow.

**Reply by hetzed** [reply]
Hi, I want to know how this is possible. Custom Queue Flow doesn't allow transfer to flow. At least I don't see a block that allows me to do that. Can you explain how can this be done?
